### ISSUE TITLE
App crash when downloading: permission denied (0.17.2)

### ISSUE THREAD

**Issue by larsgw** [TeamNewPipe/NewPipe#2634]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [x] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ GB
* __Service:__ none
* __Version:__ 0.17.1
* __OS:__ Linux motorola/athene/athene:7.0/NPJ25.93-14/16:user/release-keys 7.0 - 24

<details><summary><b>Crash log</b></summary><p>

```
java.lang.RuntimeException: java.io.IOException: Permission denied
	at us.shandian.giga.service.DownloadManager.startMission(DownloadManager.java:211)
	at us.shandian.giga.service.DownloadManagerService.startMission(DownloadManagerService.java:437)
	at us.shandian.giga.service.DownloadManagerService.lambda$onStartCommand$0(DownloadManagerService.java:207)
	at us.shandian.giga.service.-$$Lambda$DownloadManagerService$raq6lDhb_Z_YwA4Qq9oug-ZgyOM.run(lambda)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
Caused by: java.io.IOException: Permission denied
	at java.io.UnixFileSystem.createFileExclusively0(Native Method)
	at java.io.UnixFileSystem.createFileExclusively(UnixFileSystem.java:280)
	at java.io.File.createNewFile(File.java:948)
	at us.shandian.giga.service.DownloadManager.startMission(DownloadManager.java:208)
	... 10 more
java.io.IOException: Permission denied
	at java.io.UnixFileSystem.createFileExclusively0(Native Method)
	at java.io.UnixFileSystem.createFileExclusively(UnixFileSystem.java:280)
	at java.io.File.createNewFile(File.java:948)
	at us.shandian.giga.service.DownloadManager.startMission(DownloadManager.java:208)
	at us.shandian.giga.service.DownloadManagerService.startMission(DownloadManagerService.java:437)
	at us.shandian.giga.service.DownloadManagerService.lambda$onStartCommand$0(DownloadManagerService.java:207)
	at us.shandian.giga.service.-$$Lambda$DownloadManagerService$raq6lDhb_Z_YwA4Qq9oug-ZgyOM.run(lambda)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)

```
</p></details>
<hr>

 I saw #1544 but that should be fixed in `0.17` according to the linked issue.

**Comment by snappyapple632** [comment]
I strongly recommend updating to 0.17.2 or wait for 0.17.3 that will be released soon, then see if the issue is still there.

**Comment by larsgw** [comment]
I'll check with 0.17.2, it isn't on FDroid yet though.

**Comment by kapodamy** [comment]
you probably dismissed the permission ask popup. Go to android settings and grant to NewPipe the storage permission

**Comment by larsgw** [comment]
It's in Dutch, but that should be the storage permission. Either way, I would have had to dismiss it thrice, on separate occasions: once when installing the first version, a second time when re-installing it to test if something went wrong with permissions somehow (maybe an update?) and once just now when installing a newer version from the APK (which wasn't treated as an update, since I use FDroid normally), with clean settings.

![image](https://user-images.githubusercontent.com/14018963/64992751-d7ff2600-d8d4-11e9-9b41-8645054b6a8d.png)

**Comment by kapodamy** [comment]
¿have you tried activating the option "**Use SAF**" under download settings? After that you have to re-select the download directories.
![Screenshot_1568769716](https://user-images.githubusercontent.com/16857233/65095833-1fe78100-d999-11e9-8dfc-0e9e2ccba2d6.png)

**Comment by larsgw** [comment]
No luck, sorry.

<img src="https://user-images.githubusercontent.com/14018963/65175442-2a665100-da53-11e9-8cf7-e2a229a017da.png" width=400 />

**Comment by kapodamy** [comment]
try with "Ask where to download" option checked, if fails post the **crash log**

**Comment by larsgw** [comment]
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ GB
* __Service:__ none
* __Version:__ 0.17.2
* __OS:__ Linux motorola/athene/athene:7.0/NPJ25.93-14/16:user/release-keys 7.0 - 24


<details><summary><b>Crash log</b></summary><p>

```
java.lang.RuntimeException: java.io.IOException: Permission denied
	at us.shandian.giga.service.DownloadManager.startMission(DownloadManager.java:211)
	at us.shandian.giga.service.DownloadManagerService.startMission(DownloadManagerService.java:437)
	at us.shandian.giga.service.DownloadManagerService.lambda$onStartCommand$0(DownloadManagerService.java:207)
	at us.shandian.giga.service.-$$Lambda$DownloadManagerService$raq6lDhb_Z_YwA4Qq9oug-ZgyOM.run(lambda)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
Caused by: java.io.IOException: Permission denied
	at java.io.UnixFileSystem.createFileExclusively0(Native Method)
	at java.io.UnixFileSystem.createFileExclusively(UnixFileSystem.java:280)
	at java.io.File.createNewFile(File.java:948)
	at us.shandian.giga.service.DownloadManager.startMission(DownloadManager.java:208)
	... 10 more
java.io.IOException: Permission denied
	at java.io.UnixFileSystem.createFileExclusively0(Native Method)
	at java.io.UnixFileSystem.createFileExclusively(UnixFileSystem.java:280)
	at java.io.File.createNewFile(File.java:948)
	at us.shandian.giga.service.DownloadManager.startMission(DownloadManager.java:208)
	at us.shandian.giga.service.DownloadManagerService.startMission(DownloadManagerService.java:437)
	at us.shandian.giga.service.DownloadManagerService.lambda$onStartCommand$0(DownloadManagerService.java:207)
	at us.shandian.giga.service.-$$Lambda$DownloadManagerService$raq6lDhb_Z_YwA4Qq9oug-ZgyOM.run(lambda)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)

```
</p></details>
<hr>

**Comment by kapodamy** [comment]
@larsgw i see now the problem. By default the download metadata is stored in the internal memory (the path is `/sdcard/Android/data/org.schabi.newpipe[.debug]/files/pending_downloads`). 
tldr is fixable
